### PR TITLE
[MB-16449] Fix the Total Segment Count displayed in the SE segment

### PR DESCRIPTION
**Pull request by pearl-truss**
## Summary

This PR resolves an issue with the total segment count being off. I found that the invoice generator was using a hardcoded value of 7 to represent the `ServiceItemSegments` however, because the `FA2` field was previously updated to be an array, the 7 is no longer was always accurate. 

I was curious why the invoice service test didn't fail previously and I believe it's because the test code generated a single FA2 segment, so the hardcoded value of 7 was still accurate in that case. So for this PR, I created a test that generated multiple FA2 segments to check that both segments would be included in the total. 

This PR also updates how the total segment count is calculated to take into account the possibility of multiple FA2 segments

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

Please refer to the [Acceptance Testing doc](https://transcom.github.io/mymove-docs/docs/backend/testing/acceptance-testing-syncada-edi-invoicing) for EDI generation or more information re: `prime-api-client` commands

1. Go to the office app and Login as a TIO with gbloc KKFA
2. Search for move code "RDY4PY" and click it
3. Scroll down to payment requests, click Review, and approve 1+.
4. Refer to the [Acceptance Testing doc](https://transcom.github.io/mymove-docs/docs/backend/testing/acceptance-testing-syncada-edi-invoicing). 
5. In your terminal, if you have the json file from the doc in root dir, run `prime-api-client --insecure create-payment-request --filename ./payload.json | jq '.id,.paymentRequestNumber'` which should give you the 9 digit payment request ID for the following command
6. Run `make bin/prime-api-client bin/generate-payment-request-edi && generate-payment-request-edi --payment-request-number 6682-5285-1` replacing the end numbers with the payment request ID from previous
7. Confirm that the EDI prints in your terminal and does NOT include the the second FA2 field, as hhg SAC's empty. Note the SE count
8. Go back into the move and add HHG and NTS SACs
9. Run step 6 again and check that HHG SAC's in a new FA2 field and that the SE count takes the multiple FA2 fields into account

#### Count without SAC/multiple FA2 segments 
Screenshot 2023-07-03 at 10 29 04 AM

#### Count with SAC/multiple FA2 segments 
Screenshot 2023-07-03 at 10 30 21 AM

### CODE REVIEW

File: pkg/services/invoice/ghc_payment_request_invoice_generator_test.go [hunk 1]
```diff
@@ -441,6 +441,124 @@ func (suite *GHCInvoiceSuite) TestAllGenerateEdi() {
...
+				LinkOnly: true,
+			},
+		}
+		distanceZipSITOriginParam := factory.CreatePaymentServiceItemParams{
+			Key:     models.ServiceItemParamNameDistanceZipSITOrigin,
+			KeyType: models.ServiceItemParamTypeInteger,
+			Value:   "33",
+		}
+
+		// DOPSIT chosen for this test because it's a service item that generates LOA/FA segments
```
Comment (duncan-truss) [inline pkg/services/invoice/ghc_payment_request_invoice_generator_test.go:524]: Is this true for any shipment level service item or does it have to be accessorial or something else?

Comment (pearl-truss) [reply pkg/services/invoice/ghc_payment_request_invoice_generator_test.go:524]: @duncan-truss this is true for any shipment kind and based on this: https://github.com/transcom/mymove/blob/main/pkg/services/invoice/ghc_payment_request_invoice_generator.go#L846

Comment (duncan-truss) [reply pkg/services/invoice/ghc_payment_request_invoice_generator_test.go:524]: Thanks, that seemed like a harder one to mock so was curious if that was just what we were using in another test or some rule I didn't know about.  I knew the TAC/SAC won't change between service items as it's based on the shipment.

**Review by duncan-truss — APPROVED** [review]
Thanks for digging into this! I think this approach is flexible if we ever end up with some service items with a different number of FA2 segments.  For now I think they all get the same number of segments, 1 if it includes the TAC, 2 if it includes the TAC and SAC.

**Pull request merged**